tasks.php

```diff
@@ -896,17 +897,11 @@ function handle_action_on_a_specified_task()
 /**
  * Add or remove a related task to the current task.
  */
-function process_related_task($pre_task, $action, $related_task_id)
+function process_related_task(object $pre_task, string $action, int $related_task_id): void
 {
     global $pguser, $tasks_url;
     assert($action == 'add' || $action == 'remove');
-
-    // Validate task_id. It must be an integer >= 1
-    $related_task_id = trim($related_task_id);
-    if (!is_numeric($related_task_id) || $related_task_id < 1) {
-        ShowError(_("You must supply a valid related task ID."), true);
-        return;
-    }
+    assert($related_task_id >= 1);
```

Comment:
This isn't going to do what you think it will. Asserts are only valid for development, we can't rely on them for non-development purposes (see https://www.php.net/manual/en/function.assert.php -- `zend.assertions` != 1).

Since we want to validate that the id is >= 1 we have to do this in a non-assert check.

Reply:
This makes sense (and yeah, I missed the fact that asserts are disabled unless we set them up in our ini file).

One question here: we already validate that the task id is >= 1 when we call `get_{integer|float}_param`. Is it worth keeping the assert then? How about the extra validation on every user request?